redisvl/index/index.py

```diff
@@ -317,6 +321,36 @@ class SearchIndex(BaseSearchIndex):
 
     """
 
+    @classmethod
+    def from_existing(
+        cls,
+        name: str,
+        redis_client: Optional[redis.Redis] = None,
+        redis_url: Optional[str] = None,
+        **kwargs,
+    ):
+        # Handle redis instance
+        if redis_url:
+            redis_client = RedisConnectionFactory.connect(
+                redis_url=redis_url, use_async=False, **kwargs
+            )
+        if not redis_client:
+            raise ValueError(
+                "Must provide either a redis_url or redis_client to fetch Redis index info."
+            )
+
+        # Validate modules
+        installed_modules = unpack_redis_modules(
+            convert_bytes(redis_client.module_list())
+        )
+        validate_modules(installed_modules, [{"name": "search", "ver": 20810}])
```

Comment:
Is search the only one we need to validate? I'm looking at the remote possibility that somebody did their own module add in the configuration with mismatch versions, but that likely entail us to have a compatibility matrix with at least 3 columns, Redis version , Search module version, and JSON module version

Reply:
For the ability to read in from the FT.INFO output, it's not tied to any module besides Search. Not sure we can invest in supporting additional/ad-hoc modules (yet). Will likely require a bit of work to get there. But I do want to clean up the module and connection factory interfaces soon to make them a bit cleaner and more generic.